tests/test_api.rs

```diff
@@ -126,6 +126,28 @@ fn global_handles() {
   assert!(g5.is_empty());
 }
 
+#[test]
+fn global_handle_drop() {
+  let _setup_guard = setup();
+
+  // Global 'g1' will be dropped _after_ the Isolate has been disposed.
+  let mut g1 = v8::Global::<v8::String>::new();
+
+  let mut params = v8::Isolate::create_params();
+  params.set_array_buffer_allocator(v8::new_default_allocator());
+  let mut isolate = v8::Isolate::new(params);
+
+  let mut hs = v8::HandleScope::new(&mut isolate);
+  let scope = hs.enter();
+
+  let l1 = v8::String::new(scope, "foo").unwrap();
+  g1.set(scope, l1);
+
+  // Global 'g2' will be dropped _before_ the Isolate has been disposed.
+  let l2 = v8::String::new(scope, "bar").unwrap();
+  let _g2 = v8::Global::new_from(scope, l2);
+}
```

Comment:
This test ought to use drop(isolate) to really test the V8 segfault we're attempting to avoid.

Reply:
No test cases here show globals being accessed after isolate is destroyed.

Reply:
> This test ought to use drop(isolate) to really test the V8 segfault we're attempting to avoid.

No. Rust drops variables opposite of declaration order. This is guaranteed.

Reply:
> No test cases here show globals being accessed after isolate is destroyed.

The type system doesn't allow that, so you're gonna have to live with it.

Reply:
https://doc.rust-lang.org/book/ch15-03-drop.html

"Rust automatically called drop for us when our instances went out of scope, calling the code we specified. Variables are dropped in the reverse order of their creation, so d was dropped before c. This example gives you a visual guide to how the drop method works; usually you would specify the cleanup code that your type needs to run rather than a print message."

Reply:
If you put a global in a box, store it into a static pointer, can you not get it to outlive the isolate long enough to hit the V8 segfault?